src/app/researcher/study/request/[memberSlug]/study-proposal.tsx

```diff
@@ -28,23 +111,17 @@ export const StudyProposal: React.FC<{ memberId: string }> = ({ memberId }) => {
             })
             router.push(`/researcher/dashboard`)
         },
-        onError(error) {
-            // TODO server action max filesize error doesn't propagate through here...
+        onError: async (error, _, context: { studyId: string; studyJobId: string } | undefined) => {
             console.error(error)
-            notifications.show({ message: String(error), color: 'red' })
-        },
-    })
+            if (!context) return
 
-    const studyProposalForm = useForm<StudyProposalFormValues>({
-        validate: zodResolver(studyProposalSchema),
-        validateInputOnBlur: true,
-        initialValues: {
-            title: '',
-            piName: '',
-            irbDocument: null,
-            descriptionDocument: null,
-            agreementDocument: null,
-            codeFiles: [],
+            const deleteStudyFilesURL = await onDeleteStudyAction({
+                memberSlug: memberSlug,
+                studyId: context.studyId,
+                studyJobId: context.studyJobId,
+            })
+
+            await fetch(deleteStudyFilesURL)
```

Comment:
have you tested the fetch(deleteStudyFilesURL) removes all the study files?  as I recall, S3 requires looping over each file in a path and won't delete everything  in one call like this.  

It also occurs to me that we could fire the deletes from the server inside the `onDeleteStudyAction` but that doesn't matter too much

Reply:
oh right - because they're not sending over a giant payload ... good idea!